### DISCUSSION TITLE
It seems that merge_page corrupts the file, but I can't determine if it actually corrupts the file.

### DISCUSSION THREAD

**Discussion by ssjkamei** [py-pdf/pypdf#2683] [Q&A] [answered]
The files were merged using merge_page.
I am not familiar with PDF specifications, so any advice on what is broken in the file would be very helpful.

* Adobe Acrobat shows that the merged file is smaller and rotated relative to the base.
* The base is a file created with `WeasyPrint 61.2` (PDF-1.7).
* The file to be merged is the file created by `reportlab 4.2.0` (PDF-1.3).

The following is a simple code.

```python
from pypdf import PdfReader, PdfWriter

with open("merge_base.pdf", "rb") as data_f:
    merge_base = PdfReader(data_f)

    with open("add_file.pdf", "rb") as data_f:
        add_file = PdfReader(data_f)

        pdf_file_writer = PdfWriter()
        for i in range(0, 1):
            pdf_file_writer.add_page(merge_base.pages[i])

        for i in range(0, 1):
            pdf_file_writer.pages[i].merge_page(
                add_file.pages[i])

        pdf_file_writer.write("JoinedPDF.pdf")

```

[merge_base.pdf](https://github.com/py-pdf/pypdf/files/15455368/merge_base.pdf)
[add_file.pdf](https://github.com/py-pdf/pypdf/files/15455370/add_file.pdf)
[JoinedPDF.pdf](https://github.com/py-pdf/pypdf/files/15455371/JoinedPDF.pdf)

**Comment by stefan6419846** [comment, marked as answer]
Which version of *pypdf* are you using? Did you try with the latest release? I just tested it with (nearly) the latest head and rendering works correct with some minified code:

```python3
from pypdf import PdfReader, PdfWriter

merge_base = PdfReader("merge_base.pdf")
add_file = PdfReader("add_file.pdf")

pdf_file_writer = PdfWriter()
for page in merge_base.pages:
    pdf_file_writer.add_page(page)

for i in range(len(pdf_file_writer.pages)):
    pdf_file_writer.pages[i].merge_page(add_file.pages[i])

pdf_file_writer.write("JoinedPDF.pdf")
```

**Comment by ssjkamei** [comment]
I was running on `4.0.2`. It worked fine with `4.2.0`.
Thank you. Very helpful, I was struggling for about 2 days...

When upgrading from `4.0.2` to `4.1.0` or higher, I started getting errors in parallel processing using `concurrent.futures.ProcessPoolExecutor`, and I forgot I was not using the latest version.

I will register this matter for another discussion.
Thanks so much for your help.
